src/transform_all.bat

```diff
@@ -17,24 +17,30 @@ dir *.tt /b > t4list.txt
 echo the following T4 templates will be transformed:
 type t4list.txt
 
-set TextTransform=%COMMONPROGRAMFILES(x86)%\microsoft shared\TextTemplating\11.0\TextTransform.exe
+:: Use texttransform.exe from the IDE if it is present, otherwise, use legacy locations. 
 
-IF NOT EXIST "%TextTransform%" set TextTransform=%COMMONPROGRAMFILES(x86)%\microsoft shared\TextTemplating\12.0\TextTransform.exe
+:: clear text transform path to undefine it
+set TEXTTRANSFORMPATH=
 
-IF NOT EXIST "%TextTransform%" set TextTransform=%COMMONPROGRAMFILES(x86)%\microsoft shared\TextTemplating\14.0\TextTransform.exe
-
-IF NOT EXIST "%TextTransform%" set TextTransform=%ProgramFiles(x86)%\Microsoft Visual Studio\2017\Professional\Common7\IDE\TextTransform.exe
-
-IF NOT EXIST "%TextTransform%" set TextTransform=%ProgramFiles(x86)%\Microsoft Visual Studio\2017\Enterprise\Common7\IDE\TextTransform.exe
+:: use latest vswhere utility to locate in IDE, where it resides now. 
+IF EXIST "%ProgramFiles(x86)%\Microsoft Visual Studio\Installer\vswhere.exe" (
```

Comment:
do you happen to know if this always exists? Even on machines with only the vs build tools installed? 
https://visualstudio.microsoft.com/downloads/#build-tools-for-visual-studio-2019

Reply:
Yes, vswhere.exe should always be present, with just vs build tools or with VS 2015.2 Preview 2 and ahead.
This is the issue regarding the vswhere, where they decided to put it in a unified location:
*https://github.com/microsoft/vswhere/issues/41*

Reply:
great, thanks.